### PR TITLE
fix: gitlab users are not appeared in table.users

### PR DESCRIPTION
**Pull request by abeizn**
# Summary
add gitlab user collector, extractor and convertor.
update gitlab user model
update e2e test

### Does this close any open issues?
related to #2126 
related to #2268 

### Screenshots
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/gitlab/tasks/user_collector.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+func CollectUsers(taskCtx core.SubTaskContext) error {
+	rawDataSubTaskArgs, data := CreateRawDataSubTaskArgs(taskCtx, RAW_USER_TABLE)
+	logger := taskCtx.GetLogger()
+	logger.Info("collect gitlab users")
+
+	collector, err := helper.NewApiCollector(helper.ApiCollectorArgs{
+		RawDataSubTaskArgs: *rawDataSubTaskArgs,
+		ApiClient:          data.ApiClient,
+		UrlTemplate:        "/projects/{{ .Params.ProjectId }}/members",
```
Comment (likyh) [inline plugins/gitlab/tasks/user_collector.go:46]: Maybe users are not limited in members. Is this how it is planned?

Comment (abeizn) [reply plugins/gitlab/tasks/user_collector.go]: yes, I think everyone who is invited is an organizer, but the organizer determined by the company still needs to be determined by the company itself

File: plugins/gitlab/models/migrationscripts/archived/user.go [hunk 2]
```diff
@@ -20,9 +20,16 @@ package archived
 import "github.com/apache/incubator-devlake/models/migrationscripts/archived"
 
 type GitlabUser struct {
-	ConnectionId uint64 `gorm:"primaryKey"`
-	Email        string `gorm:"primaryKey;type:varchar(255)"`
-	Name         string `gorm:"type:varchar(255)"`
+	ConnectionId    uint64 `gorm:"primaryKey"`
+	ProjectId       int    `gorm:"primaryKey"`
```
Comment (klesh) [inline plugins/gitlab/models/migrationscripts/archived/user.go:24]: This should be a Many-to-Many relationship

File: plugins/gitlab/models/user.go [hunk 3]
```diff
@@ -17,12 +17,21 @@ limitations under the License.
...
+import (
+	"github.com/apache/incubator-devlake/models/common"
+)
 
 type GitlabUser struct {
-	ConnectionId uint64 `gorm:"primaryKey"`
-	Email        string `gorm:"primaryKey;type:varchar(255)"`
-	Name         string `gorm:"type:varchar(255)"`
+	ConnectionId    uint64 `gorm:"primaryKey"`
+	ProjectId       int    `gorm:"primaryKey"`
```
Comment (klesh) [inline plugins/gitlab/models/user.go:26]: Same as above

File: plugins/gitlab/models/migrationscripts/archived/user.go (hunk 2 above)
Comment (abeizn) [reply plugins/gitlab/models/migrationscripts/archived/user.go:24]: I don't understand very well, like jira, tapd and gitlab should all be unified, right? This definition structure should be able to achieve all cases

File: plugins/gitlab/tasks/user_collector.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+			return query, nil
+		},
+
+		ResponseParser: func(res *http.Response) ([]json.RawMessage, error) {
+			var result json.RawMessage
+			err := helper.UnmarshalResponse(res, &result)
+			if err != nil {
+				return nil, err
+			}
+			return []json.RawMessage{result}, nil
```
Comment (klesh) [inline plugins/gitlab/tasks/user_collector.go:62]: This doesn't seem right, by convention, we store `items` of a `page`, not the raw response body.
Even though it works in this particular case, it breaks the convention.

File: plugins/gitlab/tasks/user_extractor.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		RawDataSubTaskArgs: *rawDataSubTaskArgs,
+		Extract: func(row *helper.RawData) ([]interface{}, error) {
+			var userRes []models.GitlabUser
+			err := json.Unmarshal(row.Data, &userRes)
+			if err != nil {
+				return nil, err
+			}
+
+			results := make([]interface{}, 0)
+			for _, v := range userRes {
```
Comment (klesh) [inline plugins/gitlab/tasks/user_extractor.go:48]: We should be expecting a single User in this callback, not all Users

**Review by klesh — COMMENTED** [review]
I think we extract the Author/Assignees as User of the Issue as well.
https://github.com/apache/incubator-devlake/blob/42f6c3237d65e36bf009cec987aacbbc4b233d71/plugins/gitlab/tasks/issue_extractor.go#L54
https://github.com/apache/incubator-devlake/blob/42f6c3237d65e36bf009cec987aacbbc4b233d71/plugins/gitlab/tasks/issue_extractor.go#L65

File: plugins/gitlab/tasks/user_extractor.go (hunk 5 above)
Comment (abeizn) [reply plugins/gitlab/tasks/user_extractor.go]: updated

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**